networkx/algorithms/causal/m_separation.py

```diff
@@ -1,21 +1,35 @@
+from copy import deepcopy
+from collections import deque
+
 import networkx as nx
+from networkx.utils import UnionFind
+
+import graphs
 
 __all__ = ["m_separated"]
 
 
+def markov_blanket(
+    G, x, bidirected_edge_name="bidirected", directed_edge_name="directed"
+):
+    pass
+
+
 def m_separated(
     G, x, y, z, bidirected_edge_name="bidirected", directed_edge_name="directed"
 ):
     """Check m-separation among 'x' and 'y' given 'z' in mixed-edge causal graph G.
 
-    This algorithm wraps ``networkx.algorithms.d_separated``, but
-    allows one to pass in a ``ADMG`` instance instead.
+    This algorithm adapts the linear time algorithm presented in [1] currently implemeted in
+    networkx.algorithms.d_separation to work for mixed-edge causal graphs.
+
+    This algorithm first obtains the ancestral subgraph of x | y | z which only requires knowledge of
```

Comment:
What does `x | y | z` mean here? Is that the union of only those nodes and gets the subgraph consisting only of the ancestors of that union of nodes?

Reply:
Yep that's right. I would write X \cup Y \cup Z in latex, this particular chunk is actually copied verbatim from the current d-sep implementation